src/Methods/Pipes/Managers.php

```diff
@@ -23,25 +26,38 @@ public function handle(PassableContract $passable, Closure $next): void
         $found = false;
 
         if ($classReflection->isSubclassOf(Manager::class) && ! $classReflection->isAbstract()) {
-            $driver = null;
-
             $concrete = $this->resolve(
                 $classReflection->getName(),
             );
 
-            try {
-                $driver = $concrete->driver();
-            } catch (InvalidArgumentException) {
-                // ..
+            $class = null;
+
+            $createDefaultDriverMethod = 'create' . Str::studly($concrete->getDefaultDriver()) . 'Driver';
+            if ($classReflection->hasNativeMethod($createDefaultDriverMethod)) {
+                $createDefaultDriverMethod = $classReflection->getNativeMethod($createDefaultDriverMethod);
+                try {
+                    $methodReturnType = ParametersAcceptorSelector::selectSingle($createDefaultDriverMethod->getVariants())->getReturnType();
+                    $class            = $methodReturnType->getObjectClassNames()[0] ?? null;
+                } catch (ShouldNotHappenException) {
+                    // ..
+                }
```

Comment:
You shouldn't need to catch the ShouldNotHappenException here:

```suggestion
                $class = ParametersAcceptorSelector::selectSingle($createDefaultDriverMethod->getVariants())
                    ->getReturnType()
                    ->getObjectClassNames()[0] ?? null;
```